### ISSUE TITLE
md-menu accessibility (role=menu, role=menuitem not there)

### ISSUE THREAD

**Issue by opadillaintegrify** [angular/material#4415]
I'm new to accessibility so I might be wrong here, but I was inspecting the DOM elements of md-menu and its md-menu-items on https://material.angularjs.org/latest/#/demo/material.components.menu and was expecting to find menu related aria attributes:

role=menu 
role=menuitem

I didn't see these anywhere.  Should they be?

**Comment by marcysutton** [comment]
It uses `aria-haspopup` to indicate the trigger button *has* an accompanying menu (which is in a separate part of the DOM), but it only goes part of the way there. The menu component needs more semantics to identify the menu items, and to bind more of a relationship between the trigger button and the menu (using `aria-controls`, since the DOM structure won't be hierarchal). One thing to pay attention to is that the markup may need to be present in the DOM at all times for assistive technology, rather than generated on user interaction (a buffer/race condition). Here is a demo that uses hierarchal DOM instead of `aria-controls` on the trigger button: http://oaa-accessibility.org/example/25/

**Comment by marcysutton** [comment]
Okay, we need to make some changes to this component to actually work as a menu. Right now, the interactive elements are all `md-button`–which have their own role and keyboard functionality (each item is reached with TAB). As a result, the wrapping `role="menuitem"` is never announced, because it is not interactive or ever reached with the keyboard. Here is what I recommend doing:
* Add `role="menu"` to `md-menu`.
* Make menu items hierarchal, so they are descendants of `role="menu"`.
*  To preserve the HTML API, move `role="menuitem"` from `md-menu-item` to `md-button` (replacing `role="button"`). `md-menu-item` no longer needs a role.
* Using [roving tabindex](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets#Technique_1_Roving_tabindex) technique, add `tabindex="-1"` to `md-button` elements, with `tabindex="0"` on only one item at a time (the current one, after being selected..otherwise default to the first item)
* If developer uses a positive tabindex value, use that instead of `0`.

**Comment by marcysutton** [comment]
I forgot that Menus have to support links, something that isn't supported by the ARIA menu/menuitem roles (nor do they support text inputs in menus, as another example). For that reason, Menubar should behave more like Menu, using `aria-haspopup="true"`. Once the menu items are hierarchal children of the menu at load time, it will have the required semantics (so `aria-owns` won't be required).

**Comment by antrez770** [comment]
Hi how create Accessibility menu in Angular?

**Comment by Splaktar** [comment]
@antrez770 please try asking on [Stack Overflow](https://stackoverflow.com/), [Gitter](https://gitter.im/), or the [AngularJS Material forums](https://groups.google.com/forum/?pli=1#!forum/ngmaterial).
